### PR TITLE
fix(platform-browser): DomAdapter should resolve event key by Event.key

### LINKED ISSUE #8273: (keyup.enter) doesn't trigger in IE10
**Current behavior**

When you bind `(keyup.enter)` to an element it doesn't trigger in IE10.

**Other information**

I tried to create a plnkr from the issues and it doesn't run in IE10. 

http://plnkr.co/edit/YhxMmgqL9YKcFF2Qm035?p=preview

### LINKED ISSUE #9419: DOM helper getKeyEvent does not use keyCode
I'm not sure if this is a bug with Angular or ChromeDriver yet, but here goes for the record:

Angular's browser dom adapter uses `event.key` in `getKeyEvent`. When WebDriver via ChromeDriver sends a `sendKeys` event, it fills out the event's `keyCode`, but not the `key`. This means that something such as `<button (keydown.arrowRight)="doTheThing()">` will not be triggered in an e2e test.

cc @robertmesserle

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #8273
Closes #9419

### CODE REVIEW

File: modules/@angular/platform-browser/src/browser/browser_adapter.ts [hunk 1]
```diff
@@ -307,28 +326,37 @@ export class BrowserDomAdapter extends GenericBrowserDomAdapter {
...
-      // Safari cf
-      // http://www.w3.org/TR/2007/WD-DOM-Level-3-Events-20071221/events.html#Events-KeyboardEvents-Interfaces
-      if (isBlank(key)) {
-        return 'Unidentified';
-      }
-      if (key.startsWith('U+')) {
-        key = String.fromCharCode(parseInt(key.substring(2), 16));
-        if (event.location === DOM_KEY_LOCATION_NUMPAD && _chromeNumKeyPadMap.hasOwnProperty(key)) {
-          // There is a bug in Chrome for numeric keypad keys:
-          // https://code.google.com/p/chromium/issues/detail?id=155654
```
Comment (DzmitryShylovich) [inline modules/@angular/platform-browser/src/browser/browser_adapter.ts]: bug was fixed

**Review by mhevery — APPROVED** [review]

**Review by mhevery — CHANGES_REQUESTED** [review]
We can't take code from other projects without their approval. This can not be merged.

**Comment by IgorMinar** [comment]
we shouldn't even try to fix this PR. stealing code is not OK.

@DzmitryShylovich please do not ever try to submit PRs to any of our repos that contain stolen code. Doing so violates the Google CLA you signed and also violates license of the project you stole the code from. If such code was accidentally merged into our repo, the whole project could be jeopardized and I'm sure that's not something you'd like to see happen.

Open source is great way to learn about software and collaborate, but just because some code is open sourced it doesn't mean that it can be copy&pasted into other open source project without fulfilling all the attribution, licensing and any other legal requirements of the originating and receiving project.

**Comment by DzmitryShylovich** [comment]
@IgorMinar 
Probably `steal` was a poor word choice. I would call it `adopting` :)
Here we are talking about 20 LOC method to get event keys. It's not super secret/important algo or something like that. 
I don't really want to reinvent the wheel for such a common use case. And I don't see anything bad in adopting experience of another successful framework. And I'm 99% sure that react guys wouldn't be against. But I understand that it violates licensing so I'll try to roll out more independent solution.

**Pull request closed without merging**